src/components/CodeView/utils.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { RefractorNode } from 'refractor';
+import makeClassName from 'classnames';
+
+const mapChild = (
+  child: RefractorNode,
+  i: number,
+  depth: number,
+): React.ReactElement | string => {
+  if (child.tagName) {
+    // @ts-ignore
+    const className = makeClassName(child.properties.className);
+
+    return React.createElement(
+      child.tagName,
+      Object.assign({ key: `code-child-${depth}-${i}` }, child.properties, {
+        className,
+      }),
+      // eslint-disable-next-line no-use-before-define, @typescript-eslint/no-use-before-define
+      child.children && child.children.map(mapWithDepth(depth + 1)),
+    );
+  }
+
+  return child.value;
+};
+
+export const mapWithDepth = (depth: number) => {
```

Comment:
This and `mapChild` doesn't appear to have test coverage

Reply:
According to codecov, this is tested a it's used by the component. Can I add specific test cases in a follow-up?

Reply:
I filed: https://github.com/mozilla/addons-code-manager/issues/299